testgen/generators/common.js

```diff
@@ -7,10 +15,32 @@ function sample_tests(all_tests, run_limit) {
   }
 
   let size_all = all_tests.length;
-  let increment = Math.floor(size_all / run_limit);
-  let samples = [];
-  for (let index = 0; index < size_all; index += increment) {
-    samples.push(all_tests[index]);
+  label_increment = Math.floor(size_all / run_limit);
+  max_count = run_limit;
+  added_count = 0;
+
+  if (debug) {
+    console.log('COMMON: ', size_all, ' increment: ', label_increment,
+                ' max_count: ', max_count);
+  }
+
+  let samples = all_tests.filter(filter_every_nth);
+  if (debug) {
+    console.debug('  COMMON results: ', samples.length);
+    console.debug('  COMMON results: ', samples);
```

Comment:
I think this code would be cleaner if you just used `(i) => i % label_increment == 0` directly as the argument to `.filter()`

You don't even use `added_count` for debugging purposes because you can already get it from `samples.length`, which you do. Therefore, we can get rid of `filter_every_nth` altogether.

Reply:
Changed to (e, i) => i % label_increment == 0

where e is the element and i is the index.